### ISSUE TITLE
Simplifiying polygons

### ISSUE THREAD

**Issue by gktval** [NetTopologySuite/NetTopologySuite#675]
### Description
I am not sure if this is a bug, or if I am doing something wrong. I am trying to simplify a bunch of touching polygons, and at the same time, preserve their geometry. I have a mutlipolygon with 5 features.  If you look closely, the shape has many corners from the resulting raster it was derived from. I am trying to reduce the number of points, but keep the polygon shapes by preserving the touching boundaries.

![image](https://user-images.githubusercontent.com/7605046/224850888-cb05c5b8-f129-4c76-bd1c-b8d3fa9f43a0.png)

To simplify, I am unioning the 5 features together, then passing the resulting multipolygon into the TopologyPreservingSimplifier. The resulting polygon has places where the geometry of the multipolygons do not line up. Is this a bug in TopologyPreservingSimplifier or is there a better function to accomplish this task?

![image](https://user-images.githubusercontent.com/7605046/224851031-730a7adf-287a-4fb8-8334-1c2f7731a495.png)

### System Configuration
Tell us about the environment where you are experiencing the bug:
- NetTopologySuite version: 2.5.0
- Environment (Operating system, version and so on): Windows 11
- .NET Framework version: Net Framework 4.7.2

**Comment by airbreather** [comment]
This is a surprisingly challenging problem, and I have yet to see an off-the-shelf product (whether in NTS or otherwise) that solves it in a satisfactory way.

I haven't come up with any better ideas than "find the least significant point in the entire set of polygons, then delete it (unless deleting it would cause problems), then keep going until you've done enough".

**Comment by gktval** [comment]
Definitely not easy. Perhaps I misunderstood the purpose of TopologyPreservingSimplifier (or it is for much simpler polygons that I supplied). If I have to right something up, it's mot too big of a deal. Perhaps if it is halfway decent I share it. 

The question now becomes: Is it easier/better to create a geometry preserving polygon simplifier? or to create a clean function that removes overlaps and holes between adjacent polygons? ...

**Comment by FObermaier** [comment]
There is good and a not so good news for you
* [Polygonal coverages](http://lin-ear-th-inking.blogspot.com/2022/07/polygonal-coverages-and-operations-in.html) are already implemented 
* [Simplyfing polygonal coverages](http://lin-ear-th-inking.blogspot.com/2023/03/simplifying-polygonal-coverages-with-jts.html) is not yet implemented

**Comment by FObermaier** [comment]
@gktval, with the latest pre-release package you should be ready to go:

```C#
double threshold; // simplification threshold from somewhere
NetTopologySuite.Geometries.Geometry[] coverageGeoms; // from somewhere

// if you want to simplify everything
NetTopologySuite.Geometries.Geometry[] simplifiedGeoms = 
    NetTopologySuite.Coverage.CoverageSimplifier.Simplify(coverageGeoms, threshold);

// if you want to simplify just the interior of the coverage and leave the boundary untouched
NetTopologySuite.Geometries.Geometry[] simplifiedGeoms = 
    NetTopologySuite.Coverage.CoverageSimplifier.SimplifyInner(coverageGeoms, threshold);
```

**Comment by gktval** [comment]
Awesome! Thanks, I will give it a go. I will close this in the meantime.

**Comment by FObermaier** [comment]
> Awesome! Thanks, I will give it a go. I will close this in the meantime.

I'm interested in the outcome of this

**Comment by gktval** [comment]
> I'm interested in the outcome of this

I'll be gone this week. So it may be a couple weeks before I can report back. Just FYI in case you will be waiting.

**Comment by gktval** [comment]
I was just getting back around to this after a few years. Sorry, things got busy and I was working on other projects. I installed the latest 2.6.0 package, but did not see the Coverage namespace. Am I missing it somewhere?

**Comment by gktval** [comment]
@FObermaier Took me a few years to get back to this project. I see you just released NTS 2.6.0. The tool works like a charm. Reduce my vertex count from 20000 down to 3500. Thank you so much for putting this together.

![Image](https://github.com/user-attachments/assets/13c014f8-eb4b-4706-acd7-8f0983378e6f)
